x12-edi/2.2/modules/ROOT/pages/index.adoc

```diff
@@ -0,0 +1,597 @@
+|interchangeId |Interchange identifier, if errorLevel INTERCHANGE_LEVEL or above.
+|groupId |Group identifier, if errorLevel GROUP_LEVEL or TRANSACTION_LEVEL.
+|transactionId |Interchange identifier, if errorLevel TRANSACTION_LEVEL.
+|===
+
+Error data is returned by the read operation as optional lists with the "Errors" key at every level of the data structure. At the transaction set level, this list contains non-fatal errors encountered during the parsing of that transaction set. At the interchange level, this list contains errors (both fatal and non-fatal) for transaction sets with fatal errors. At the root level of the read, this list contains both interchange errors and every other error reported at any nested level.
+
+== To Write X12 EDI Messages
+
+To write an outgoing message, search the palette for X12 EDI and drag the X12 Write operation into a flow. Construct an outgoing X12 EDI message according to the structure as defined above for input to the write operation. If no errors are found in the write operation the normal flow continues. Otherwise, an exception is thrown which includes an array of X12Error instances, the same as returned by the Read operation (documented above).
+
+== To Send Functional Acknowledgments
+
+Sending generated functional acknowledgments (997 or 999 transaction sets) is the same as writing any other EDI message, except you set the transactions to the acknowledgments that were generated during the read operation.
+
+For example:
+
+image::x12-edi-ack-flow.jpg[Ack Flow Diagram]
+
+[source,xml,linenums]
+----
+<x12-edi:read config-ref="X12_EDI__Configuration" doc:name="Read EDI Doc"/>
+  ...
+<dw:transform-message doc:name="Create Outgoing Acks Message">
+    <dw:set-payload><![CDATA[%dw 1.0
+%output application/java
+---
+{
+	TransactionSets: {
+		v005010: {
+			'997' : payload.FunctionalAcksGenerated
+		}
+	}
+}]]></dw:set-payload>
+</dw:transform-message>
+<x12-edi:write config-ref="X12_EDI__Configuration" doc:name="Send Acks"/>
+----
+
+The generated functional acknowledgments have interchange data set up for sending back to the sender of the original message, so you don't need to change anything in the transactions to do the send.
+
+If you use 999 acknowledgments, add any required CTX segments yourself to the basic structure generated by the X12 Module. The X12 connector does not track implementation convention changes to the base transaction set structure.
+
+== Sending Interchange Acknowledgments
+
+To send interchange acknowledgments (TA1 segments), set the value of the InterchangeAcksToSend key in the message map to the list of TA1 segment maps to be sent. The generated TA1 segments have interchange data set up for sending back to the sender of the original message, so you don't need to change anything in the segment data to do the send.
+
+== See Also
+
+* xref:release-notes::connector/x12-edi-connector-release-notes-mule-4.adoc[X12 EDI Connector Release Notes]
```

Comment:
Some of the See Also links don't seem to be links.

Reply:
check